addons/account_voucher/models/account_voucher.py

```diff
@@ -328,8 +328,9 @@ def voucher_move_line_create(self, line_total, move_id, company_currency, curren
                             ctx = {}
                             if self.account_date:
                                 ctx['date'] = self.account_date
-                            temp['currency_id'] = current_currency.id
-                            temp['amount_currency'] = company_currency._convert(tax_vals['amount'], current_currency, line.company_id, self.account_date or fields.Date.today(), round=True)
+                            temp['currency_id'] = current_currency
```

Comment:
why not set in the beginning of the function
company_currency = self.company_id.currency_id
current_currency = self.currency_id
and use those instead of re-browse at every iteration of the loop which is probably not efficient.
Otherwise it seems correct to me.